Backend/starter_project_mavericks/BlogApp/tests/BlogApp.UnitTests/User/SignInHandlerTests.cs

```diff
@@ -0,0 +1,207 @@
+//             Assert.Equal("mock-token", result.Data.Token);
+//         }
+
+//         [Fact]
+//         public async Task Handle_ShouldThrowException_WhenUserNotFound()
+//         {
+//             // Arrange
+//             var email = "user@example.com";
+//             var password = "Pa$$w0rd";
+//             var signinFormDto = new SigninFormDto { Email = email, Password = password };
+
+//             _userManagerMock.Setup(u => u.FindByEmailAsync(email))
+//                 .ReturnsAsync((User)null);
+
+//             var handler = new SigninHandler(_authRepository);
+
+//             // Act & Assert
+//             await Assert.ThrowsAsync<NotFoundException>(() => handler.Handle(new SigninCommand { SigninFormDto = signinFormDto }, default));
+//         }
+
+//         [Fact]
+//         public async Task Handle_ShouldThrowException_WhenInvalidCredentialsProvided()
+//         {
+//             // Arrange
+//             var email = "test1@example.com";
+//             var password = "Pa$$w0rd";
+//             var user = new User { Email = email, UserName = email, PasswordHash = "invalid-password" };
+//             var signinFormDto = new SigninFormDto { Email = email, Password = password };
+
+//             _userManagerMock.Setup(u => u.FindByEmailAsync(email))
+//                 .ReturnsAsync(user);
+//             _signInManagerMock.Setup(s => s.CheckPasswordSignInAsync(user, password, false))
+//                 .ReturnsAsync(Microsoft.AspNetCore.Identity.SignInResult.Failed);
+
+//             var handler = new SigninHandler(_authRepository);
+
+//             // Act & Assert
+//             await Assert.ThrowsAsync<BadRequestException>(() => handler.Handle(new SigninCommand { SigninFormDto = signinFormDto }, default));
+//         }
+//     }
+
+using BlogApp.Application.Features.Authentication.DTO;
+using BlogApp.Application.Models.Identity;
+using BlogApp.Application.Exceptions;
+using BlogApp.Identity.Repositories;
+using BlogApp.Domain;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Moq;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Xunit;
+using BlogApp.Persistence;
+using Microsoft.EntityFrameworkCore;
+
+namespace BlogApp.Identity.UnitTests
+{
+    public class SignInHandlerTests
+    {
+        private readonly AuthRepository _authRepository;
+        private readonly Mock<UserManager<User>> _userManagerMock;
+        private readonly Mock<SignInManager<User>> _signInManagerMock;
+
+        public SignInHandlerTests()
+        {
+            var services = new ServiceCollection();
+            var configuration = new ConfigurationBuilder()
+                .AddJsonFile("/home/bahailu/Documents/a2sv/Internship-2023/Backend/starter_project_mavericks/BlogApp/BlogApp.API/appsettings.json", optional: true, reloadOnChange: true)
```

Comment:
THe folder path provided here should relative not absolute

Reply:
When I use relative path I am getting the following error 

Error Message:
   System.IO.FileNotFoundException : The configuration file '../../../BlogApp.API/appsettings.json' was not found and is not optional.
  Stack Trace:
     at Microsoft.Extensions.Configuration.FileConfigurationProvider.Load(Boolean reload)
   at Microsoft.Extensions.Configuration.ConfigurationRoot..ctor(IList`1 providers)
   at Microsoft.Extensions.Configuration.ConfigurationBuilder.Build()
   at BlogApp.Identity.UnitTests.AuthServiceTests..ctor() in /home/bahailu/Documents/a2sv/Internship-2023/Backend/starter_project_mavericks/BlogApp/tests/BlogApp.UnitTests/User/SignUpHandlerTests.cs:line 37
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)